variant-box/pi-setup.sh

```diff
@@ -75,7 +75,65 @@ sudo zip -j   -u /boot/secrets.zip   /var/lib/tor/ssh/hostname
 
 
 # TODO: detect `-source` suffix in `bitcoind-version`; if present build from source, if not build from binaries
-# TODO: how to detect arch for binaries?
+BITCOINVER=`cat bundle/bitcoind-version`
+if [[ $BITCOINVER == *"source"* ]]; then 
+    echo "Building bitcoin from source"
+else
+    echo "Using pre-built binary"
+    # Double check if its actually a supported platform
+    # TODO: use my detect arch script as fallback
+    if [ $(uname -m) == "armv7l" ]; then
+        sudo apt-get install -y \
+            apt-transport-https \
+            ca-certificates \
+            curl \
+            software-properties-common
+
+        # verify key
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+        sudo apt-key fingerprint 0EBFCD88
+
+        echo "deb [arch=armhf] https://download.docker.com/linux/debian \
+            $(lsb_release -cs) stable" | \
+            sudo tee /etc/apt/sources.list.d/docker.list
+
+        # update sources
+        sudo apt-get update
+
+        # install docker
+        sudo apt-get install -y docker-ce
+
+        # setup docker containers
+        sudo usermod -G docker pi # Make sure the permissions are set
+        sudo systemctl start docker # Make sure docker is started
+
+        # create folder structure
+        mkdir -p /home/pi/data/btc
```

Comment:
I think docker installation should be separate from bitcoind and LN node. Let's move make it to always happen **before** bitcoin version check?